### ISSUE TITLE
Is policy network using communication at the code? I thought image feature and comm feature were just used.

### ISSUE THREAD

**Issue by DH-O** [ToruOwO/marl-ae-comm#2]
First of all, thank you for your paper and code. It is really helpful to my research.

According to ae.py file, there is a code
`cat_feat[i] = torch.cat([cat_feat[i], comm_out[i].unsqueeze(0)],
                                    dim=-1)`

why there is a comm_out[i].unsqueeze? According to the paper, The GRU policy (actually, the code you uploaded uses LSTM) uses concatenates the encoded image features and the message features which is cat_feat[i].

Thank you for your response.

**Comment by ToruOwO** [comment]
Thank you for your interest! In theory, comm_out only contains redundant
features that are already represented by image features and messages
features so shouldn't be needed; in practice we find it stabilises training
so still included it. Hope that answers your question.

**Comment by DH-O** [comment]
Thank you 😊
